cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/common/MockEmitter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.common;
+
+import co.cask.cdap.templates.etl.api.Emitter;
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+
+/**
+ * Mock emitter for unit tests
+ */
+public class MockEmitter<K, V> implements Emitter<K, V> {
+  private final Map<K, V> emitted = Maps.newHashMap();
```

Comment:
Technically this should be List? I can emit multiple but identical <K, V> pairs. I guess it is fine for unit tests?